aptos-move/framework/src/natives/cryptography/groups/arkworks_backend.rs

```diff
@@ -0,0 +1,1896 @@
+    gas_params: &GasParameters,
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    assert_eq!(1, ty_args.len());
+    let type_tag = context
+        .type_to_type_tag(ty_args.get(0).unwrap())?
+        .to_string();
+    let bytes = pop_arg!(args, Vec<u8>);
+    match type_tag.as_str() {
+        "0x1::groups::BLS12_381_G1" => {
+            let cost = gas_params.bls12_381.g1_affine_deserialize_uncompressed + gas_params.bls12_381.g1_affine_to_proj;
+            let point = ark_bls12_381::G1Affine::deserialize_uncompressed(bytes.as_slice());
+            match point {
+                Ok(point) => {
+                    let handle = context
+                        .extensions_mut()
+                        .get_mut::<Bls12381Context>()
+                        .add_g1_point(point.into_projective());
+                    Ok(NativeResult::ok(
+                        cost,
+                        smallvec![Value::bool(true), Value::u64(handle as u64)],
+                    ))
+                }
+                _ => {
+                    Ok(NativeResult::ok(
+                        cost,
+                        smallvec![Value::bool(false), Value::u64(0)],
+                    ))
+                }
+            }
+        }
+        "0x1::groups::BLS12_381_G2" => {
+            let cost = gas_params.bls12_381.g2_affine_deserialize_uncompressed + gas_params.bls12_381.g2_affine_to_proj;
+            let point = ark_bls12_381::G2Affine::deserialize_uncompressed(bytes.as_slice());
+            match point {
+                Ok(point) => {
+                    let handle = context
+                        .extensions_mut()
+                        .get_mut::<Bls12381Context>()
+                        .add_g2_point(point.into_projective());
+                    Ok(NativeResult::ok(
+                        cost,
+                        smallvec![Value::bool(true), Value::u64(handle as u64)],
+                    ))
+                }
+                _ => {
+                    Ok(NativeResult::ok(
+                        cost,
+                        smallvec![Value::bool(false), Value::u64(0)],
+                    ))
+                }
+            }
+        }
+        "0x1::groups::BLS12_381_Gt" => {
+            let cost = (gas_params.bls12_381.fq12_deserialize + gas_params.bls12_381.fq12_pow_fr + gas_params.bls12_381.fq12_eq) * NumArgs::one();
+            let point = Fq12::deserialize(bytes.as_slice());
+            match point {
+                Ok(point) => {
+                    let r_buf = hex::decode(
+                        "01000000fffffffffe5bfeff02a4bd5305d8a10908d83933487d9d2953a7ed73",
+                    )
+                        .unwrap();
+                    let r =
+                        ark_ff::BigInteger256::deserialize_uncompressed(r_buf.as_slice()).unwrap();
+                    if Fq12::one() == point.pow(r) {
```

Comment:
So this will be sufficient to check membership in `GT` because of two things:
1. GT is a multiplicative subgroup of `Fq12`, which is a finite field
2. All multiplicative subgroups of finite fields are cyclic (e.g., see [this](http://math.stanford.edu/~conrad/210BPage/handouts/math210b-finite-mult-groups-cyclic.pdf) or [this](https://math.stackexchange.com/questions/837562/why-is-the-multiplicative-group-of-a-finite-field-cyclic))
3. If there exists a subgroup of order d in a cyclic group of order n, then that subgroup is unique (see [this](https://en.wikipedia.org/wiki/Subgroups_of_cyclic_groups) or [this](https://math.stackexchange.com/questions/378521/subgroups-of-cyclic-groups-with-same-order))